### PR TITLE
separate anaconda build from anaconda worker commands

### LINKED ISSUE #94: Refactor build worker cli
I think the cli should be split into worker and build eg
```
anaconda worker [run|register|deregister|list]

anaconda build [init|submit|trigger,...etc]
```

### PR DESCRIPTION
**Pull request by PeterDSteinberg**
Fixes #94, refactoring build worker cli. 
Commands begin with one of:
```anaconda build```
```anaconda worker```
```anaconda docker_worker```

In the repo, the commands directory moved to build_commands, and the worker_commands and docker_worker_commands directories were created.  The three *_commands directories separate the three CLI usages above.

Here is the new sequence of commands in use:

```bash
anaconda worker -h
anaconda worker register psteinberg8/abc # register it to my abc queue
# it should print out a worker id like '561d506614b9ec3fc8e6f09c' 
anaconda worker run 561d506614b9ec3fc8e6f09c
# finally when done, deregister the worker
anaconda worker deregister 561d506614b9ec3fc8e6f09c
# at any point, check for information about the workers
anaconda worker list

# and the same pattern exists for docker_worker

anaconda docker_worker -h
anaconda docker_worker register psteinberg8/abc # register it to my abc queue
# it should print out a worker id like '561d506614b9ec3fc8e6f09c' 
anaconda docker_worker run 561d506614b9ec3fc8e6f09c
# finally when done, deregister the worker
anaconda docker_worker deregister 561d506614b9ec3fc8e6f09c
# at any point, check for information about the workers
anaconda docker_worker list
```

### CODE REVIEW

**Comment by srossross** [comment]
One small change. Can we do `anaconda worker docker-run` instead of `anaconda docker-worker`?

**Comment by PeterDSteinberg** [comment]
That makes sense and will help reduce the number of directories/files.

**Comment by srossross** [comment]
Can you add an error message when a user runs `anaconda build worker`?

**Comment by PeterDSteinberg** [comment]
The commit I just pushed includes an error message if using ```anaconda build worker``` and it also changes ```anaconda docker_worker run``` to ```anaconda worker docker_run <worker_id>```

File: setup.py [hunk 1]
```diff
@@ -18,12 +18,15 @@
...
-          'console_scripts': [
-              'binstar-build = binstar_build_client.scripts.build:main',
-              'conda-clean-build-dir = binstar_build_client.scripts.conda_clean_build_dir:main',
-              ],
-           'conda_server.subcommand': ['build = binstar_build_client.scripts.build:add_parser'],
-                 },
+     entry_points={
+      'console_scripts': [
+          'binstar-build = binstar_build_client.scripts.build:main',
+          'binstar-worker = binstar_build_client.scripts.worker:main',
```
Comment (srossross) [inline setup.py]: I don't think this command is required because the `binstar-*` commands are for legacy only.

**Comment by srossross** [comment]
@PeterDSteinberg it looks like only the python 2 tests are failing. (I fixed the build output by running the worker in python3).

**Comment by srossross** [comment]
@PeterDSteinberg what is left on this PR?

**Comment by PeterDSteinberg** [comment]
It looks to be failing because of stdout, readline, and the reader thread, not because of anything specific to this PR.  I think we should work on a fix to the bugs in this test in a separate branch.

File: binstar_build_client/build_commands/worker.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import argparse
+
+from binstar_build_client.worker_commands.run import add_parser as add_parser_run
+
+
+def main(args):
+    print(__doc__)
+
+def add_parser(subparsers, name='worker',
+               description='DEPRECATED',
```
Comment (srossross) [inline binstar_build_client/build_commands/worker.py]: Can you add more of a description here?

The cli output from `-h` is only
```
    worker       DEPRECATED
```

**Comment by srossross** [comment]
> I think we should work on a fix to the bugs in this test in a separate branch

I agree

Can you fix the `description='DEPRECATED'` above then I give it a :+1:

**Comment by srossross** [comment]
:+1:

**Comment by malev** [comment]
@PeterDSteinberg please remove the branch after merging

**Pull request merged**